### ISSUE TITLE
[Discussion] UI/UX Reflow

### ISSUE THREAD

**Issue by Gitoffthelawn** [einaregilsson/Redirector#289]
[This recent interesting feature request](https://github.com/einaregilsson/Redirector/issues/288) got me thinking *(uh oh)*.

The primary UX issue I have with Redirector is that it requires multiple steps to toggle on/off.  There are currently 2 methods to accomplish this task, and both require multiple steps:

1. The first method is to click on the toolbar icon, and then click on the "Disable Redirector" (or "Enable Redirector") button.  Personally, I keep Redirector in Firefox's overflow menu, so I also have the added step of opening the overflow menu.
2. The second method involved a little clever trickery.  You first create a shortcut to the internal page that contains the "Disable Redirector" (or "Enable Redirector") button.  You can then create a bookmark keyword to that page to open it.  To toggle Redirector, you use that keyword to access the page, and then click the "Disable Redirector" (or "Enable Redirector") button.  I actually find this faster than the first method, plus it gives you a temporary browser tab to toggle Redirector to help you remember to turn it back on.

But what if, instead of requiring multiple steps, there was a way to toggle Redirector with just a single click?

How about a toolbar button that simply toggles Redirector, and another to edit redirects and change settings?

Everything would stay the way it is, except the Redirector button would finally toggle Redirector and a new `Edit Redirects` (or `Configure Redirector`) button would be added that goes to the current `Edit Redirects` page that would now also have the `Enable logging` and `Enable notification` options.

_[As our dear friend Linda would say, talk amongst yourselves.](https://www.youtube.com/watch?v=XVloHiUCKyg)_

**Comment by einaregilsson** [comment]
Hmmm, don't the browsers restrict each addon to one toolbar button? Or am I misremembering things?

**Comment by Gitoffthelawn** [comment]
> Hmmm, don't the browsers restrict each addon to one toolbar button? Or am I misremembering things?

Hi Einar,

Hmmm hmmm, that's a very interesting point.  Ya know, you might be right.  If true, that's a silly and arbitrary browser limitation, but one that would be challenging to work around.

There used to be a way for the user to create new buttons for extensions, but I think all those holes have been plugged.  Well, actually, the user can still inject user scripts into the browser itself, but I've never completely understood all the risks involved, so I choose not to do that sort of thing.

If your memory is correct (and I hope it isn't, but I think it may be), I can think of 2 other possibilities:

1. At least in Firefox, extensions can have context menu actions for their toolbar buttons.  Stylus and uBlock Origin (uBO) extensions both do this.  So the Redirector button could toggle Redirector, and one could access the rule editor via the Redirector toolbar button's context menu.
2. I wonder if a bookmarklet could be created to toggle Redirector.

**Comment by einaregilsson** [comment]
Yeah, I think mostly people don't want an extension to take up too much real estate in the browser menu, two buttons is a lot. I'll think about it :)

**Comment by Gitoffthelawn** [comment]
Thanks Einar.  Redirector has become one of those extensions I can't imagine browsing without.  I haven't recently counted the number of redirect rules I currently employ, but I'm guessing it's between 50 and 100.
